x/concentrated-liquidity/lp.go

```diff
@@ -37,16 +37,39 @@ func (k Keeper) createPosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddr
 		return sdk.Int{}, sdk.Int{}, sdk.Dec{}, err
 	}
 
-	// calculate liquidity created from this position
+	// N.B. we only write cache context if actual amounts
+	// returned are greater than the given minimums.
+	cacheCtx, writeCacheCtx := ctx.CacheContext()
```

Comment:
Let's move this to be right before the call to `updatePosition`. I don't see a reason to be initializing this before validation checks that can fail

Reply:
I see not that this is being used by `k.setPool`.

Maybe in the future PR, we can make the `updatePosition` method take `pool` instead of `poolId`. That way we wouldn't have to call `k.setPool` prior to `updatePosition`.

I think it is best to gather all mutative function in one place - `updatePosition`.

Reply:
Yeah, right now it needs to be initialized there for setting the pool for the first position. Agreed on taking pool for a future PR, I will make an issue, add it here, then resolve conversation.